examples/chip-tool/main.cpp

```diff
@@ -100,18 +100,14 @@ static bool ContentMayBeADataModelMessage(System::PacketBuffer * buffer)
 static void HandleDataModelMessage(System::PacketBuffer * buffer)
 {
     EmberApsFrame frame;
-    bool ok = extractApsFrame(buffer->Start(), buffer->DataLength(), &frame);
-    if (ok)
-    {
-        printf("APS frame processing success!\n");
-    }
-    else
+    if (extractApsFrame(buffer->Start(), buffer->DataLength(), &frame) == 0)
     {
         printf("APS frame processing failure!\n");
         System::PacketBuffer::Free(buffer);
         return;
     }
 
+    printf("APS frame processing success!\n");
     uint8_t * message;
     uint16_t messageLen = extractMessage(buffer->Start(), buffer->DataLength(), &message);
```

Comment:
I would have expected this to look more like:
```
uint8_t * message = buffer->Start() + offset;
uint16_t messageLen = buffer->DataLength() - offset;
```
or something, where `offset` is the return value of `extractApsFrame`.  I guess you're leaving `extractMessage` but just reimplementing it on top of `extractApsFrame`?  But if we do that, do we want to change the return type, or have a static non-public `doExtractApsFrame` helper that returns the offset and is called by both `extractApsFrame`, which returns whether the offset is nonzero, and by `extractMessage`?

The tradeoff of doing things the way they're done here is we do all the `extractApsFrame` work twice.  Maybe that's ok?

Reply:
Spoke offline. The cost of using `extractApsFrame` is minimal. So we are keeping it this way for now.